generated/nidcpower/nidcpower/lcr_load_compensation_spot.py

```diff
@@ -0,0 +1,145 @@
+import ctypes
+import numbers
+import platform
+
+import nidcpower._visatype
+import nidcpower.enums as enums
+
+
+# This class is an internal implementation detail
+# ctypes definition
+# Name must match exactly what the name of the structure type is named in the C API.
+class struct_NILCRLoadCompensationSpot(ctypes.Structure):  # noqa N801
+    if platform.system() == "Windows" and platform.architecture()[0] == "64bit":
+        _pack_ = 8
+    elif platform.system() == "Linux" or platform.system() == "Darwin":
+        _pack_ = 4
+    else:
+        _pack_ = 1
+    _fields_ = [
+        ("frequency", nidcpower._visatype.ViReal64),
+        ("reference_value_type", nidcpower._visatype.ViInt32),
+        ("reference_value_a", nidcpower._visatype.ViReal64),
+        ("reference_value_b", nidcpower._visatype.ViReal64),
+    ]
+
+    def __init__(self, data=None):
+        super(ctypes.Structure, self).__init__()
+        if data is not None:
+            self.frequency = data.frequency
+            self.reference_value_type = enums.LCRReferenceValueType(data.reference_value_type).value
+            if isinstance(data, LCRLoadCompensationSpot):
+                if self.reference_value_type == enums.LCRReferenceValueType.IMPEDANCE.value:
+                    self.reference_value_a = data.reference_value.real
+                    self.reference_value_b = data.reference_value.imag
+                else:
+                    self.reference_value_a = data.reference_value
+                    self.reference_value_b = 0.0
+            else:
+                self.reference_value_a = data.reference_value_a
+                self.reference_value_b = data.reference_value_b
+        else:
+            # Assign default value for all fields
```

Comment:
I think Python/ctypes may get you this for free.

Reply:
As part of removing the unnecessary code paths in https://github.com/ni/nimi-python/pull/1702#discussion_r920326129, these default settings are no longer needed in my latest commit.